### ISSUE TITLE
Compile-AppInNavContainer with dependency generates an error

### ISSUE THREAD

**Issue by mynavblog** [microsoft/navcontainerhelper#526]
Dear Freddy,

I am trying to use **Compile-AppInNavContainer** in project where I have the dependency in the Application (App.json) but I get strange error then:

```
ConvertFrom-Json : Invalid JSON primitive: Â.
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.6.2.6\AppHandling\Compile-AppInNavContainer.ps1:132 char:60
+ ... ppJsonObject = Get-Content -Raw -Path $appJsonFile | ConvertFrom-Json

    + CategoryInfo          : NotSpecified: (:) [ConvertFrom-Json], ArgumentException
    + FullyQualifiedErrorId : System.ArgumentException,Microsoft.PowerShell.Commands.ConvertFromJsonCommand
```

when deleting the dependency it works fine (mean this particular project I cannot compile because I have errors related to missing symbols but this is expected).

**Comment by freddydk** [comment]
Sounds like a bug - will investigate

**Comment by freddydk** [comment]
Just tried this:
```
$auth = "NavUserPassword"
$credential = New-Object pscredential 'admin', (ConvertTo-SecureString -String 'P@ssword1' -AsPlainText -Force)
$containerName = "bc"
$imageName = "mcr.microsoft.com/businesscentral/onprem:1904"

New-BcContainer -accept_eula `
                -accept_outdated `
                -containerName $containerName `
                -imageName $imageName `
                -auth $auth `
                -Credential $credential `
                -updateHosts `
                -licenseFile $licenseFile `
                -includeTestToolkit `
                -includeTestLibrariesOnly

$app1ProjectFolder = "c:\programdata\navcontainerhelper\bc-app"
Compile-AppInBCContainer -containerName $containerName -credential $credential -appProjectFolder $app1ProjectFolder -appOutputFolder $app1ProjectFolder -appName "bc1.app" -UpdateSymbols
Publish-NavContainerApp -containerName $containerName -appFile (Join-Path $app1ProjectFolder "bc1.app") -skipVerification -sync -install

$app2ProjectFolder = "c:\programdata\navcontainerhelper\bc2-app"
Compile-AppInBCContainer -containerName $containerName -credential $credential -appProjectFolder $app2ProjectFolder -appOutputFolder $app2ProjectFolder -appName "bc2.app" -UpdateSymbols
Publish-NavContainerApp -containerName $containerName -appFile (Join-Path $app2ProjectFolder "bc2.app") -skipVerification -sync -install
```
That works.

Can you elaborate on what you are doing? Potentially repro steps?

**Comment by mynavblog** [comment]
Hi,
Thx for answer.

Do you have in App.json file dependency between app1 and app2? 

```
 "dependencies": [
    {
      "appId": "My Id",
      "name": "My Name",
      "publisher": "My Publisher",
      "version": "2.0.0.0"
    }
  ],
```

I run it on image: mcr.microsoft.com/businesscentral/onprem:13.4.28874.29613-dk

I simply run:

```
$credential = New-Object pscredential 'admin', (ConvertTo-SecureString -String 'P@ssword1' -AsPlainText -Force)


Compile-AppInBCContainer -containerName bcbuild -credential $credential -appProjectFolder "C:\Users\myuser\Documents\AL\App1" -appOutputFolder "C:\Users\myuser\Documents\AL" -UpdateSymbols
```
Best regards,
Krzysztof

**Comment by freddydk** [comment]
Yes, my dependency is from bc2-app to bc-app (obviously), which is why I compile and publish bc-app first and then bc2-app can get symbols, compile and publish.
my bc2-app json file has:

  "dependencies": [
    {
      "appId": "cb99d78b-f9db-4a1e-822a-0c9c444535df",
      "name": "Hello World",
      "publisher": "Microsoft",
      "version": "1.0.0.0"
    }
  ],

What is the exact content of your dependency?
Any special characters?

**Comment by mynavblog** [comment]
I am trying to reproduce the problem on the simpler (as your example). I have the special character in the Publisher Name for example: **Hello A/S**

But now I face issue:
```
Invoke-RestMethod : Could not find a part of the path 'C:\Users\user\Documents\AL\BC2\.alpackages\Hello A\S_hello 250 and hello 32_1.0.0.0.app'.
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.6.2.6\AppHandling\Compile-AppInNavContainer.ps1:265 char:13
+             Invoke-RestMethod -Method Get -Uri $url @AuthParam -OutFi ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Invoke-RestMethod], DirectoryNotFoundException
    + FullyQualifiedErrorId : System.IO.DirectoryNotFoundException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
```
I run the code:
```
$credential = New-Object pscredential 'admin', (ConvertTo-SecureString -String 'P@ssword1' -AsPlainText -Force)
Compile-AppInBCContainer  -credential $credential -containerName bcbuild  -appProjectFolder "C:\Users\user\Documents\AL\BC2" -appOutputFolder "C:\Users\user\Documents\AL\BC2" -UpdateSymbols
```
and here is app.json:

```
{
  "id": "f7cc19ea-4df4-4298-8461-f4c23e003864",
  "name": "Hello 360 Second",
  "publisher": "XXX NewCompany Danmark A/S",
  "version": "2.0.0.0",
  "brief": "",
  "description": "description description.",
  "privacyStatement": "",
  "EULA": "",
  "help": "",
  "url": "https://www.dxc.technology",
  "logo": "Resources/DXCLogo.png",
  "dependencies": [
    {
      "appId": "c73da900-7dd8-4f93-a364-21f8fd381712",
      "name": "Hello 360 New",
      "publisher": "XXX NewCompany Danmark A/S",
      "version": "1.0.0.0"
    }
  ],
  "screenshots": [],
  "platform": "13.0.0.0",
  "application": "13.0.0.0",
  "idRange":  {"from": 50000, "to": 60000},
  "showMyCode": true,
  "runtime": "2.3",
  "target": "Internal",
  "features": ["TranslationFile"]
}
```

**Comment by mynavblog** [comment]
I am trying to reproduce the problem on the simpler (as your example). I have the special character in the Publisher Name for example: **Hello A/S**

But now I face issue:
```
Invoke-RestMethod : Could not find a part of the path 'C:\Users\user\Documents\AL\BC2\.alpackages\Hello A\S_hello 250 and hello 32_1.0.0.0.app'.
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.6.2.6\AppHandling\Compile-AppInNavContainer.ps1:265 char:13
+             Invoke-RestMethod -Method Get -Uri $url @AuthParam -OutFi ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Invoke-RestMethod], DirectoryNotFoundException
    + FullyQualifiedErrorId : System.IO.DirectoryNotFoundException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
```
I run the code:
```
$credential = New-Object pscredential 'admin', (ConvertTo-SecureString -String 'P@ssword1' -AsPlainText -Force)
Compile-AppInBCContainer  -credential $credential -containerName bcbuild  -appProjectFolder "C:\Users\user\Documents\AL\BC2" -appOutputFolder "C:\Users\user\Documents\AL\BC2" -UpdateSymbols
```
and here is app.json:

```
{
  "id": "f7cc19ea-4df4-4298-8461-f4c23e003864",
  "name": "Hello 360 Second",
  "publisher": "XXX NewCompany Danmark A/S",
  "version": "2.0.0.0",
  "brief": "",
  "description": "description description.",
  "privacyStatement": "",
  "EULA": "",
  "help": "",
  "url": "https://www.xxx.xxx",
  "logo": "Resources/DXCLogo.png",
  "dependencies": [
    {
      "appId": "c73da900-7dd8-4f93-a364-21f8fd381712",
      "name": "Hello 360 New",
      "publisher": "XXX NewCompany Danmark A/S",
      "version": "1.0.0.0"
    }
  ],
  "screenshots": [],
  "platform": "13.0.0.0",
  "application": "13.0.0.0",
  "idRange":  {"from": 50000, "to": 60000},
  "showMyCode": true,
  "runtime": "2.3",
  "target": "Internal",
  "features": ["TranslationFile"]
}
```

**Comment by mynavblog** [comment]
I am trying to reproduce the problem on the simpler (as your example). I have the special character in the Publisher Name for example: **XXX NewCompany Danmark A/S**

But now I face issue:
```
Invoke-RestMethod : Could not find a part of the path 'C:\Users\user\Documents\AL\BC2\.alpackages\Hello A\S_hello 250 and hello 32_1.0.0.0.app'.
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.6.2.6\AppHandling\Compile-AppInNavContainer.ps1:265 char:13
+             Invoke-RestMethod -Method Get -Uri $url @AuthParam -OutFi ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Invoke-RestMethod], DirectoryNotFoundException
    + FullyQualifiedErrorId : System.IO.DirectoryNotFoundException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
```
I run the code:
```
$credential = New-Object pscredential 'admin', (ConvertTo-SecureString -String 'P@ssword1' -AsPlainText -Force)
Compile-AppInBCContainer  -credential $credential -containerName bcbuild  -appProjectFolder "C:\Users\user\Documents\AL\BC2" -appOutputFolder "C:\Users\user\Documents\AL\BC2" -UpdateSymbols
```
and here is app.json:

```
{
  "id": "f7cc19ea-4df4-4298-8461-f4c23e003864",
  "name": "Hello 360 Second",
  "publisher": "XXX NewCompany Danmark A/S",
  "version": "2.0.0.0",
  "brief": "",
  "description": "description description.",
  "privacyStatement": "",
  "EULA": "",
  "help": "",
  "url": "https://www.xxx.xxx",
  "logo": "Resources/XXXLogo.png",
  "dependencies": [
    {
      "appId": "c73da900-7dd8-4f93-a364-21f8fd381712",
      "name": "Hello 360 New",
      "publisher": "XXX NewCompany Danmark A/S",
      "version": "1.0.0.0"
    }
  ],
  "screenshots": [],
  "platform": "13.0.0.0",
  "application": "13.0.0.0",
  "idRange":  {"from": 50000, "to": 60000},
  "showMyCode": true,
  "runtime": "2.3",
  "target": "Internal",
  "features": ["TranslationFile"]
}
```

**Comment by freddydk** [comment]
Ok, that might be the missing piece.
No national characters?
Will look into this.

**Comment by mynavblog** [comment]
not at all. I even deleted the translation file

**Comment by freddydk** [comment]
Have found 2 bugs on this thanks.
a fix will be included in a new containerhelper tomorrow, which should work.
I have also changed my tests to have a publisher with a / and a name with unicode characters.

**Comment by freddydk** [comment]
Fixed in 0.6.2.9

**Comment by mynavblog** [comment]
Tested and now working all fine. Thx Freddy!!!
